### DISCUSSION TITLE
New Relic Agent Support

### DISCUSSION THREAD

**Discussion by thiagobenine** [pyinstaller/pyinstaller#6662] [Help]
I have a script that I run using the [New Relic Agent](https://docs.newrelic.com/docs/apm/agents/python-agent/getting-started/introduction-new-relic-python/). Is it possible to use PyInstaller with this agent ?

**Comment by bwoodsend** [comment]
Use it how? PyInstaller and this Relic thing look like two completely unrelated tools.

**Reply by thiagobenine** [reply]
From my understanding, `PyInstaller` generates an `.exe` file that will call the actual `python-interpreter` to run `my_script.py`. The problem is that I dont do `python my_script.py`, what I do in order to execute the script is `new-relic-agent my_script.py`. Can PyInstaller, in some way, call `new-relic-agent` instead of `python` to run `my_script`?

**Reply by bwoodsend** [reply]
Ahh. In that case, not in any sanity preserving way.

**Reply by bwoodsend** [reply]
If it has an alternative Python API way to plumb it in within Python code then that should be fairly doable. Something like [what coverage.py has](https://coverage.readthedocs.io/en/6.3.2/api.html#coverage-py-api) (if it exists) could just be put at the top of your main script.

**Reply by rokm** [reply]
Maybe via ["manual integration"](https://docs.newrelic.com/docs/apm/agents/python-agent/installation/python-agent-advanced-integration/#manual-integration)?

**Reply by thiagobenine** [reply]
Thanks @rokm and @bwoodsend. In the end, I just gave up and I'm using their Rest API to do what I wanted to do.
